### PR TITLE
Expand user in config paths

### LINKED ISSUE #11533: Support `~` in config
It seems like we currently do not support `~` in our config, e.g.

```python
import mne

mne.set_config("MNE_DATA", "~/Data")
mne.datasets.sample.data_path()
```

raises a `FileNotFoundError: Download location ~/Data as specified by MNE_DATA does not exist. Either create this directory manually and try again, or set MNE_DATA to an existing directory.` (the directory exists).

Is there a reason why we would not want to expand the tilde? This would be useful to store MNE data in the same location within a user's home directory across operating systems.

### PR DESCRIPTION
**Pull request by cbrnr**
Fixes #11533.

### CODE REVIEW

**Comment by cbrnr** [comment]
Are we still hitting the Nilearn issue after #11478? Do we need to add a new filter?

File: mne/datasets/utils.py [hunk 1]
```diff
@@ -98,12 +98,12 @@ def _get_path(path, key, name):
     # 3. get_config('MNE_DATA')
     path = get_config(key or 'MNE_DATA', get_config('MNE_DATA'))
     if path is not None:
-        if not op.exists(path):
+        if not op.exists(op.expanduser(path)):
```
Comment (drammock) [inline mne/datasets/utils.py:101]: why not use pathlib throughout?
```suggestion
        path = Path(path).expanduser()
        if not path.exists():
```

and then the corresponding change below

File: mne/datasets/utils.py [hunk 2]
```diff
@@ -98,12 +98,12 @@ def _get_path(path, key, name):
...
             msg = (f"Download location {path} as specified by MNE_DATA does "
                    f"not exist. Either create this directory manually and try "
                    f"again, or set MNE_DATA to an existing directory.")
             raise FileNotFoundError(msg)
-        return Path(path)
+        return Path(path).expanduser()
```
Comment (drammock) [inline mne/datasets/utils.py:106]: (the aforementioned "corresponding change")

```suggestion
        return path
```

**Review by drammock — COMMENTED** [review]
just one suggestion; done through GitHub UI so it may also require removing the `os.path` import if that's its only usage in the file.

**Comment by cbrnr** [comment]
Thanks @drammock, `op` is used in many other places, replacing that with `pathlib` would be great, but should be done in a separate PR. I think everything else looks good!

**Pull request merged**